### PR TITLE
twister: support board@revision in platform filter

### LINKED ISSUE #48386: twister cannot take `board@revision` as platform filter
**Describe the bug**
It is possible to use west build to build for different board revision, but twister does not support it.

For example, `west build -p -b nucleo_f030r8@1 samples/hello_world/` works but running `./scripts/twister -p nucleo_f030r8@1 -T samples/hello_world/` has the following error:

```
INFO    - Zephyr version: zephyr-v3.1.0-2698-gd5222f318d29
INFO    - Using 'zephyr' toolchain.
ERROR   - platform_filter - unrecognized platform - nucleo_f030r8@1
```

**Expected behavior**
Twister can using board revision in platform filter.

**Impact**
This prevents developers to test new board revisions before introducing them into the tree.

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK 0.14.2
 - Commit d5222f318d29bb49458b6f47a0c1f8ebed134d39

### PR DESCRIPTION
**Pull request by chen-png**
support to use board@revision as platform filter when running
twister, like "twister -p nucleo_f030r8@1 ...".

met a build failure on stm32f411e_disco@B board when CI tested this PR, 
so added the fix in this PR too.

Signed-off-by: Chen Peng1 <peng1.chen@intel.com>

fixes: #48386

### CODE REVIEW

**Review by stephanosio — DISMISSED** [review]
CMSIS-DSP-related changes LGTM.

**Comment by stephanosio** [comment]
@chen-png I have cherry-picked your CMSIS-DSP commits into https://github.com/zephyrproject-rtos/zephyr/pull/48475 so they can be merged quickly as hotfix.

**Comment by stephanosio** [comment]
Rebased

**Comment by chen-png** [comment]
> I have cherry-picked your CMSIS-DSP commits into #48475 so they can be merged quickly as hotfix.

thanks

**Comment by PerMac** [comment]
Do I understand it correctly: If there are multiple revisions available, all of them will be added to the platform scope? I'm worried that it can bloat the CI scope. Maybe only the latest revision should be added as default?

**Comment by nashif** [comment]
> Do I understand it correctly: If there are multiple revisions available, all of them will be added to the platform scope? I'm worried that it can bloat the CI scope. Maybe only the latest revision should be added as default?

depends on how you use revision, in some cases this can be different hw completely. How is adding a new revision is different from boards being added on a daily basis to the "scope"? 
IMO we should not discriminate at this level and any revision should be looked at as a new configuration to be tested and looked at.

File: scripts/pylib/twister/twisterlib/testplan.py [hunk 1]
```diff
@@ -366,6 +367,23 @@ def add_configurations(self):
                         self.platforms.append(platform)
                         if platform.default:
                             self.default_platforms.append(platform.name)
+                        # support board@revision
+                        # if there is already an existed <board>_<revision>.yaml, then use it to
+                        # load platform directly, otherwise, iterate the directory to
+                        # get all valid board revision based on each <board>_<revision>.conf.
+                        if not "@" in platform.name:
```
Comment (hakehuang) [inline scripts/pylib/twister/twisterlib/testplan.py:374]: can we document this?

Comment (chen-png) [reply scripts/pylib/twister/twisterlib/testplan.py:374]: I added this in the twister doc

**Review by hakehuang — APPROVED** [review]

**Comment by chen-png** [comment]
@PerMac could you help review this again? thanks.

**Review by enjiamai — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**